dagger-common/src/main/java/com/gotocompany/dagger/common/core/StencilClientOrchestrator.java

```diff
@@ -41,9 +42,29 @@ StencilConfig createStencilConfig() {
         return StencilConfig.builder()
                 .fetchHeaders(getHeaders(configuration))
                 .fetchTimeoutMs(configuration.getInteger(SCHEMA_REGISTRY_STENCIL_FETCH_TIMEOUT_MS, SCHEMA_REGISTRY_STENCIL_FETCH_TIMEOUT_MS_DEFAULT))
+                .cacheAutoRefresh(configuration.getBoolean(SCHEMA_REGISTRY_STENCIL_CACHE_AUTO_REFRESH_KEY, SCHEMA_REGISTRY_STENCIL_CACHE_AUTO_REFRESH_DEFAULT))
+                .cacheTtlMs(configuration.getLong(SCHEMA_REGISTRY_STENCIL_CACHE_TTL_MS_KEY, SCHEMA_REGISTRY_STENCIL_CACHE_TTL_MS_DEFAULT))
+                .refreshStrategy(getSchemaRefreshStrategy(configuration.getString(SCHEMA_REGISTRY_STENCIL_REFRESH_STRATEGY_KEY, SCHEMA_REGISTRY_STENCIL_REFRESH_STRATEGY_DEFAULT)))
+                .fetchBackoffMinMs(configuration.getLong(SCHEMA_REGISTRY_STENCIL_FETCH_BACKOFF_MIN_MS_KEY, SCHEMA_REGISTRY_STENCIL_FETCH_BACKOFF_MIN_MS_DEFAULT))
+                .fetchRetries(configuration.getInteger(SCHEMA_REGISTRY_STENCIL_FETCH_RETRIES_KEY, SCHEMA_REGISTRY_STENCIL_FETCH_RETRIES_DEFAULT))
                 .build();
     }
 
+    private SchemaRefreshStrategy getSchemaRefreshStrategy(String refreshStrategy) {
+        if (refreshStrategy == null) {
+            return SchemaRefreshStrategy.longPollingStrategy();
+        }
+
+        if (refreshStrategy.equalsIgnoreCase("LONG_POLLING")) {
+            return SchemaRefreshStrategy.longPollingStrategy();
+        }
+        if (refreshStrategy.equalsIgnoreCase("VERSION_BASED")) {
+            return SchemaRefreshStrategy.versionBasedRefresh();
+        }
+        LOGGER.error("Invalid Schema Refresh Strategy - {} , only LONG_POLLING or VERSION_BASED allowed", refreshStrategy);
+        throw new IllegalArgumentException("Invalid Schema Refresh Strategy - " + refreshStrategy);
```

Comment:
yeah but what if the user actually wants version_based but for example misspells it in the arguments, then it will get set to long_polling, and the user might be thinking it is running version_based. So , wouldn't it be better to throw Exception if the config value isn't valid .

Reply:
fixed